### DISCUSSION TITLE
Does it support printing?

### DISCUSSION THREAD

**Discussion by JavierDamico** [ahrm/sioyek#380] [General]
Does Sioyek support printing? I tried searching everywhere and couldn't find the option. If it doesn't, are you planning to add support?

**Comment by ahrm** [comment]
No and no.

**Comment by x61** [comment]
@ahrm, just noticed this thread and I also happened to create a related one. Could you please let us know why you do not intent to implement printing functionality? I am just wondering how you manager to print files, if needed, if sioyek is your default pdf reader. Do you use a different pdf reader for printing if needed? Just for my information.

**Reply by ahrm** [reply]
The reason for that is sioyek is a PDF *viewer*. Printing PDFs is basically the opposite of what a PDF viewer does because you print PDFs exactly when you don't want to read them on a computer.
Also another reason is that I developed sioyek because I was not satisfied with the current PDF viewer software so I had to develop it myself, however, I am satisfied with current printing software so I am not really motivated to dedicate time to this.

> I am just wondering how you manager to print files, if needed, if sioyek is your default pdf reader. Do you use a different pdf reader for printing if needed? Just for my information.

I rarely print PDFs, but when I do I use google chrome.

**Reply by x61** [reply]
Thank you for the explanation. At times, I need to print a pdf document to sign. Since sioyek is my default pdf reader, the file opens in sioyek, and find myself having to return to the download fold, and open the file in a different pdf reader to print. I thought it'll be convenient to print the file directly from within sioyek. But I do appreciate your response.

**Reply by ahrm** [reply]
Note that you can add something like this to your `prefs_user` (if `chrome.exe` is in your path):

```
new_command _open_in_chrome chrome.exe "%1"
```

Now you can open the document in chrome (or other software that you like) by executing `_open_in_chrome`. You can also bind `_open_in_chrome` to a keybinding in `keys_user.config`.

**Comment by konosubakonoakua** [comment]
If you have sumatrapdf installed, things get easy.
```yaml
# prefs_user.config
new_command _print_file sumatrapdf.exe -print-dialog -exit-when-done "%{file_path}"
```
and
```yaml
# keys_user.config
_print_file <c-p>
```
